models/demos/wormhole/llama31_8b/tt/llama_model.py

```diff
@@ -80,21 +80,95 @@ def forward(
         if mode == "prefill" and get_last_token == -1:
             return x
 
-        # Slicing the tensor to the nearest celing/floor multiples of 32 for the prefill_len, to get the last token
+        # Slicing the tensor to the nearest ceiling/floor multiples of 32 for the prefill_len, to get the last token
         if get_last_token != -1:
             x = ttnn.slice(x, (0, 0, get_last_token, 0), (1, 1, get_last_token + 32, 4096))
 
         x = self.norm(x)
+        output = self.lm_head(x)
+
+        return output
+
 
-        output = ttnn.linear(
-            x,
-            self.output_weight,
-            compute_kernel_config=self.args.compute_kernel_config_hifi4,
-            program_config=self.model_config["OUTPUT_MM_PROGCFG"],
-            memory_config=self.model_config["OUTPUT_MM_MEMCFG"],
-            dtype=self.dtype,
+class LMHead(nn.Module):
+    def __init__(
+        self,
+        args,
+        device,
+        dtype,
+        state_dict,
+        weight_cache_path,
+    ):
+        super().__init__()
+        self.args = args
+        self.device = device
+        self.dtype = dtype
+        self.vocab_size = args.vocab_size
+        self.num_splits = 4  # 2 splits hangs after a few iterations, 1 split doesn't fit into L1
+
+        split_size = self.vocab_size // self.num_splits
+
+        # Helper function to create output memory config
+        def create_output_mem_config(size):
+            padded_size = math.ceil(size / (32 * 12)) * (32 * 12)
```

Comment:
Comment why are you picking these numbers. I assume tile size and dram cores?

Reply:
Yep